Backend/app/rider-platform/rider-app/Main/src/SharedLogic/MerchantConfig.hs

```diff
@@ -79,28 +77,29 @@ updateCustomerFraudCounters riderId merchantConfigs = Redis.withNonCriticalCross
   where
     incrementCount ind = SWC.incrementWindowCount (mkCancellationKey ind riderId.getId)
 
-updateTotalRidesCounters :: (HedisFlow m r, CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, ClickhouseFlow m r) => Id Person.Person -> m ()
-updateTotalRidesCounters riderId = Redis.withNonCriticalCrossAppRedis $ do
-  _ <- getTotalRidesCount riderId
-  let key = mkTotalRidesKey riderId.getId
-  void $ Redis.incr key
+updateTotalRidesCounters :: (CacheFlow m r, MonadFlow m, EsqDBFlow m r, CacheFlow m r, EsqDBReplicaFlow m r, ClickhouseFlow m r) => Id Person.Person -> m ()
+updateTotalRidesCounters riderId = do
+  totalRidesCount <- getTotalRidesCount riderId Nothing
+  let totalRidesCount' = totalRidesCount + 1
+  void $ QP.updateTotalRidesCount riderId (Just totalRidesCount')
 
 updateTotalRidesInWindowCounters :: (CacheFlow m r, MonadFlow m) => Id Person.Person -> [DMC.MerchantConfig] -> m ()
 updateTotalRidesInWindowCounters riderId merchantConfigs = Redis.withNonCriticalCrossAppRedis $ do
   mapM_ (\mc -> incrementCount mc.id.getId mc.fraudRideCountWindow) merchantConfigs
   where
     incrementCount ind = SWC.incrementWindowCount (mkRideWindowCountKey ind riderId.getId)
 
-getTotalRidesCount :: (HedisFlow m r, CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, ClickhouseFlow m r) => Id Person.Person -> m Int
-getTotalRidesCount riderId = Redis.withNonCriticalCrossAppRedis $ do
-  let key = mkTotalRidesKey riderId.getId
-  mbTotalCount <- Redis.safeGet key
-  case mbTotalCount of
-    Just totalCount -> pure totalCount
+getTotalRidesCount :: (CacheFlow m r, MonadFlow m, EsqDBFlow m r, EsqDBReplicaFlow m r, ClickhouseFlow m r) => Id Person.Person -> Maybe DSR.SearchRequest -> m Int
+getTotalRidesCount riderId mSearchReq = do
+  totalRides <- QP.fetchTotalRides riderId
+  totalRides' <- case totalRides of
+    Nothing -> CHB.findCountByRiderIdAndStatus riderId BT.COMPLETED
+    Just totalRidesCount -> pure totalRidesCount
+  case mSearchReq of
     Nothing -> do
-      totalCount <- CHB.findCountByRiderIdAndStatus riderId BT.COMPLETED
-      Redis.setExp key totalCount 14400
-      pure totalCount
+      QP.updateTotalRidesCount riderId (Just totalRides')
+      pure totalRides'
+    Just sreq -> pure $ fromMaybe totalRides' sreq.totalRidesCount
```

Comment:
if it's there is searchReq, then finding total rides from person and clickhouse is unnecessary right ? then this should be your first check, try to reduce the number of queries.